test/WebJobs.Extensions.Tests/Extensions/NotificationHubs/NotificationHubEndToEndTests.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.NotificationHubs;
+using Microsoft.Azure.WebJobs.Extensions.NotificationHubs;
+using Microsoft.Azure.WebJobs.Extensions.Tests.Common;
+using Microsoft.Azure.WebJobs.Host;
+using Moq;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Tests.Extensions.NotificationHubs
+{
+    public class NotificationHubEndToEndTests
+    {
+        private const string AttributeConnStr = "Endpoint=sb://TestAttrNS.servicebus.windows.net/;SharedAccessKeyName=DefaultFullSharedAccessSignature;SharedAccessKey=RandomKey";
+        private const string AttributeHubName = "AttributeHubName";
+        private const string ConfigConnStr = "Endpoint=sb://TestConfigNS.servicebus.windows.net/;SharedAccessKeyName=DefaultFullSharedAccessSignature;SharedAccessKey=RandomKey";
+        private const string DefaultConnStr = "Endpoint=sb://TestNS.servicebus.windows.net/;SharedAccessKeyName=DefaultFullSharedAccessSignature;SharedAccessKey=RandomKey";
+        private const string DefaultHubName = "TestHub";
+        private const string MessagePropertiesJSON = "{\"message\":\"Hello\",\"location\":\"Redmond\"}";
+        private const string UserIdTag = "myuserid123";
+        private static Notification TestNotification = Converter.BuildTemplateNotificationFromJsonString(MessagePropertiesJSON);
+
+        [Fact]
+        public void OutputBindings()
+        {
+            //Arrage
+            var serviceMock = new Mock<INotificationHubClientService>(MockBehavior.Strict);
+            serviceMock
+                .Setup(x => x.SendNotificationAsync(It.IsAny<Notification>(), It.IsAny<string>()))
+                  .Returns(Task.FromResult(new NotificationOutcome()));
+
+            var factoryMock = new Mock<INotificationHubClientServiceFactory>(MockBehavior.Strict);
+            factoryMock
+                .Setup(f => f.CreateService(ConfigConnStr, DefaultHubName))
+                .Returns(serviceMock.Object);
+
+            var testTrace = new TestTraceWriter(TraceLevel.Warning);
+
+            //Act
+            RunTest("Outputs", factoryMock.Object, testTrace);
+
+            // Assert
+            factoryMock.Verify(f => f.CreateService(ConfigConnStr, DefaultHubName), Times.Once());
+            serviceMock.Verify(m => m.SendNotificationAsync(It.IsAny<Notification>(), It.IsAny<string>()), Times.Exactly(9));
```

Comment:
>Times.Exactly(9) [](start = 107, length = 16)

For the sake of completion, I'd have your Collectors each add at least one too. Then just bump this up.